### PR TITLE
fix: gittar remove skipAuth

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR
/kind feature

#### What this PR does / why we need it:
fix: gittar remove skipAuth

#### Which issue(s) this PR fixes:
[【gittar】鉴权取消skipAuth](https://erda.cloud/erda/dop/projects/387/issues/all?id=222904&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIl0sImFzc2lnbmVlSURzIjpbIjEwMDEyNjEiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=-1&type=TASK)
- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      gittar remove skipAuth        |
| 🇨🇳 中文    |      取消skipAuth逻辑，使用协议来区分鉴权逻辑      |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/gittar/auth/auth.go [hunk 1]
```diff
@@ -453,3 +446,11 @@ func getOrgIDV3(c *webcontext.Context, orgName string) (int64, error) {
 	}
 	return orgID, nil
 }
+
+// isGitProtocolRequest if request like git clone,git push... return true
+func isGitProtocolRequest(c *webcontext.Context) bool {
```
Comment (sfwn) [inline modules/gittar/auth/auth.go:451]: add unit test for this function.

File: modules/gittar/auth/auth.go [hunk 2]
```diff
@@ -196,50 +196,43 @@ func doAuth(c *webcontext.Context, repo *models.Repo, repoName string) {
...
+			c.AbortWithStatus(500, errors.New("the userID is empty"))
+			return
+		}
+		userInfoDto, err := uc.FindUserById(userIdStr)
+		if err != nil {
+			c.AbortWithStatus(500, err)
 			return
 		}
+		logrus.Infof("repo: %s userId: %v, username: %s", repoName, userIdStr, userInfoDto.Username)
+		//校验通过缓存5分钟结果
```
Comment (sfwn) [inline modules/gittar/auth/auth.go:219]: use english

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/gittar/auth/auth.go (hunk 2 above)
Comment (littlejiancc) [reply modules/gittar/auth/auth.go]: done

File: modules/gittar/auth/auth.go (hunk 1 above)
Comment (littlejiancc) [reply modules/gittar/auth/auth.go:451]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**